lib/src/server/server.dart

```diff
@@ -23,62 +23,77 @@ import '../shared/security.dart';
 import 'handler.dart';
 import 'service.dart';
 
+class ServerTlsCredentials {
+  final List<int> certificate;
+  final String certificatePassword;
+  final List<int> privateKey;
+  final String privateKeyPassword;
+
+  /// TLS credentials for a [Server].
+  ///
+  /// If the [certificate] or [privateKey] is encrypted, the password must also
+  /// be provided.
+  ServerTlsCredentials(
+      {this.certificate,
+      this.certificatePassword,
+      this.privateKey,
+      this.privateKeyPassword});
+
+  SecurityContext get securityContext {
+    final context = createSecurityContext(true);
+    if (privateKey != null) {
+      context.usePrivateKeyBytes(privateKey, password: privateKeyPassword);
+    }
+    if (certificate != null) {
+      context.useCertificateChainBytes(certificate,
+          password: certificatePassword);
+    }
+    return context;
+  }
+}
+
 /// A gRPC server.
 ///
 /// Listens for incoming RPCs, dispatching them to the right [Service] handler.
 class Server {
   final Map<String, Service> _services = {};
-  final int port;
-  final SecurityContext _securityContext;
 
   ServerSocket _insecureServer;
   SecureServerSocket _secureServer;
   final _connections = <ServerTransportConnection>[];
 
-  Server._(List<Service> services, this.port, this._securityContext) {
+  /// Create a server for the given [services].
+  Server(List<Service> services) {
     for (final service in services) {
       _services[service.$name] = service;
     }
   }
 
-  /// Create a server for the given [services] with no transport security,
-  /// listening on [port].
-  factory Server.insecure(List<Service> services, {int port}) {
-    return new Server._(services, port ?? 80, null);
-  }
-
-  /// Create a secure server for the given [services], listening on [port].
+  /// Return the port that the server is listening on.
   ///
-  /// If the [certificate] or [privateKey] is encrypted, the password must also
-  /// be provided.
-  factory Server.secure(List<Service> services,
-      {List<int> certificate,
-      String certificatePassword,
-      List<int> privateKey,
-      String privateKeyPassword,
-      int port}) {
-    final context = createSecurityContext(true);
-    if (privateKey != null) {
-      context.usePrivateKeyBytes(privateKey, password: privateKeyPassword);
-    }
-    if (certificate != null) {
-      context.useCertificateChainBytes(certificate,
-          password: certificatePassword);
-    }
-    return new Server._(services, port ?? 443, context);
+  /// This is only valid when the server is serving, ie. after [serve] has been
```

Comment:
Maybe better to document that `null` will be returned, if this getter is called outside the serve-to-shutdown interval. Otherwise, the getter should throw.